tcms/xmlrpc/api/testrun.py

```diff
@@ -259,20 +259,18 @@ def env_value(request, action, run_ids, env_value_ids):
     """
     from tcms.management.models import TCMSEnvValue
 
-    trs = TestRun.objects.filter(pk__in=pre_process_ids(value=run_ids))
-    evs = TCMSEnvValue.objects.filter(
+    test_runs = TestRun.objects.filter(pk__in=pre_process_ids(value=run_ids))
+    env_values = TCMSEnvValue.objects.filter(
         pk__in=pre_process_ids(value=env_value_ids)
     )
 
-    for tr in trs.iterator():
-        for ev in evs.iterator():
+    for test_run in test_runs.iterator():
+        for env_value in env_values.iterator():
             try:
-                func = getattr(tr, action + '_env_value')
-                func(env_value=ev)
+                func = getattr(test_run, action + '_env_value')
+                func(env_value=env_value)
```

Comment:
This is a rather ugly implementation (not your fault) but we can slightly improve things. I believe this calls the TestRun.add_env_value or TestRun.remove_env_value methods. remove_env_value can be refactored to use filter().delete() instead of get() && delete(). after this is done you can get rid of the except blocks like mentioned above.

Reply:
Done